lib/active_merchant/billing/gateways/credorax.rb

```diff
@@ -253,10 +254,27 @@ def add_3ds_2_optional_fields(post, options)
 
       private
 
+      def isk_change_check(money)
+        if test?
+          isk_change_time = Time.new(2021, 8, 22, 8, 0, 0, '+00:00')
+        else
+          isk_change_time = Time.new(2021, 8, 31, 8, 0, 0, '+00:00')
+        end
+
+        amount = amount(money)
+
+        if Time.now.utc >= isk_change_time
+          return sprintf('%.0f', amount.to_f / 100)
```

Comment:
This formatting looks good for amounts in `cents`, but for non-cents users of the library, this does not retain all of the logic in `localized_amount`:

https://github.com/activemerchant/active_merchant/blob/b353ff30848206540dac348420a661ca08ff377c/lib/active_merchant/billing/gateway.rb#L284-L288

I'd recommend either:
- Adding support for non-cents users of the library here, or
- It would be a bit of a refactor, but we can avoid rewriting a new formatter and leverage existing code better by overwriting the `non_fractional_currency?` method for this gateway. It currently reads:

https://github.com/activemerchant/active_merchant/blob/b353ff30848206540dac348420a661ca08ff377c/lib/active_merchant/billing/gateway.rb#L270-L272

It could simplify this change a lot to add `|| (currency is ISK AND time > isk_change_time)` to this function rather than creating a new amount formatter.

Reply:
Ahh, I didn't realize that the gateway itself determines whether it is set to `cents`, I thought it was the caller. In that case the logic is good as is - your call on if you want to switch implementation or not.

Reply:
Overwriting `non_fractional_currency?` proved complex in combination with allowing a test time value, which I think is more beneficial, so I'm going to leave it similar to this initial pattern.